src/mbgl/programs/fill_extrusion_program.cpp

```diff
@@ -0,0 +1,49 @@
+#include <mbgl/programs/fill_extrusion_program.hpp>
+#include <mbgl/sprite/sprite_atlas.hpp>
+#include <mbgl/style/cross_faded_property_evaluator.hpp>
+#include <mbgl/tile/tile_id.hpp>
+#include <mbgl/map/transform_state.hpp>
+
+namespace mbgl {
+
+using namespace style;
+
+static_assert(sizeof(FillExtrusionLayoutVertex) == 12, "expected FillExtrusionLayoutVertex size");
+
+FillExtrusionPatternUniforms::Values
+FillExtrusionPatternUniforms::values(mat4 matrix,
+                                     const SpriteAtlasElement& a,
+                                     const SpriteAtlasElement& b,
+                                     const Faded<std::string>& fading,
+                                     const UnwrappedTileID& tileID,
+                                     const TransformState& state,
+                                     const float heightFactor,
+                                     const std::array<float, 3>& lightColor,
```

Comment:
Can this be a `Color`, `lightPos` be a `Position`? Or could all three of `lightColor`, `lightPos`, and `lightIntensity` be passed as one `Light::Evaluated` value?